File: web/src/api/order.js
import request from '@/utils/request'
export function fetchList(params) {
  return request({
    url:'/order/list',
    method:'get',
    params:params
  })
}

export function closeOrder(data) {
  return request({
    url:'/order/closeOrders',
    method:'post',
    data:data
  })
}

export function deleteOrder(data) {
  return request({
    url:'/order/delete',
    method:'delete',
    data:data
  })
}

export function cancelOrder(data) {
  return request({
    url:'/order/cancelOrder',
    method:'post',
    data:data
  })
}
export function deliveryOrder(data) {
  return request({
    url:'/order/update/delivery',
    method:'post',
    data:data
  });
}

export function getOrderDetail(params) {
  return request({
    url:'/order/detail',
    method:'get',
    params: params,
  });
}

export function updateReceiverInfo(data) {
  return request({
    url:'/order/update/receiverInfo',
    method:'post',
    data:data
  });
}

export function updateMoneyInfo(data) {
  return request({
    url:'/order/update/moneyInfo',
    method:'post',
    data:data
  });
}

export function updateOrderNote(data) {
  return request({
    url:'/order/update/note',
    method:'post',
    data:data
  })
}

export function updateOrderCompletedStatus(data) {
  return request({
    url:'/order/update/complete',
    method:'post',
    data:data
  })
}

export function getOrderSetting(params) {
  return request({
    url:'/order/setting',
    method:'get',
    params:params
  })
}

export function updateOrderSetting(data) {
  return request({
    url:'/order/settingUpdate',
    method:'post',
    data:data
  });
}

File: server/router/wechat/order.go
package wechat

import (
	v1 "cooller/server/api/v1"
	"cooller/server/middleware"
	"github.com/gin-gonic/gin"
)

type OrderRouter struct{}

func (s *OrderRouter) InitOrderRouter(Router *gin.RouterGroup) (R gin.IRoutes) {
	wxOrderRouter := Router.Group("order").Use(middleware.OperationRecord())
	wxOrderApi := v1.ApiGroupApp.WechatApiGroup.OrderApi
	{
		wxOrderRouter.POST("generateConfirmOrder", wxOrderApi.GenerateConfirmOrder)
		wxOrderRouter.POST("generateOrder", wxOrderApi.GenerateOrder)
		wxOrderRouter.GET("detail", wxOrderApi.GetOrderDetail)
		wxOrderRouter.GET("list", wxOrderApi.GetOrderList)
		wxOrderRouter.POST("paySuccess", wxOrderApi.PaySuccess)
		wxOrderRouter.POST("cancelOrder", wxOrderApi.CancelOrders)
		wxOrderRouter.GET("setting", wxOrderApi.GetOrderSetting)
		wxOrderRouter.POST("settingUpdate", wxOrderApi.UpdateOrderSetting)
		wxOrderRouter.POST("closeOrders", wxOrderApi.CloseOrders)
		wxOrderRouter.DELETE("delete", wxOrderApi.DeleteOrders)
		wxOrderRouter.POST("update/receiverInfo", wxOrderApi.UpdateOrderReceiverInfo)
		wxOrderRouter.POST("update/moneyInfo", wxOrderApi.UpdateOrderMoneyInfo)
		wxOrderRouter.POST("update/note", wxOrderApi.UpdateOrderNote)
		wxOrderRouter.POST("update/complete", wxOrderApi.UpdateOrderCompletedStatus)
		wxOrderRouter.POST("cart", wxOrderApi.CreateProductCart)
		wxOrderRouter.PUT("cart", wxOrderApi.UpdateProductCartQuantity)
		wxOrderRouter.DELETE("cart", wxOrderApi.DeleteProductCartById)
		wxOrderRouter.DELETE("cart/clear", wxOrderApi.ClearProductCart)
		wxOrderRouter.GET("cart/list", wxOrderApi.GetProductCartList)
		wxOrderRouter.DELETE("carts", wxOrderApi.DeleteProductCartByIds)
		wxOrderRouter.POST("tmpCart", wxOrderApi.CreateProductTmpCart)
	}
	return wxOrderRouter
}
